Source/Core/Core/HW/WiimoteCommon/DataReport.cpp

```diff
@@ -0,0 +1,425 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <cassert>
+
+#include "Core/HW/WiimoteCommon/DataReport.h"
+
+namespace WiimoteCommon
+{
+bool DataReportManipulator::HasIR() const
+{
+  return 0 != GetIRDataSize();
+}
+
+bool DataReportManipulator::HasExt() const
+{
+  return 0 != GetExtDataSize();
+}
+
+u8* DataReportManipulator::GetDataPtr()
+{
+  return data_ptr;
+}
+
+const u8* DataReportManipulator::GetDataPtr() const
+{
+  return data_ptr;
+}
+
+struct IncludeCore : virtual DataReportManipulator
+{
+  bool HasCore() const override { return true; }
+
+  void GetCoreData(CoreData* result) const override
+  {
+    *result = *reinterpret_cast<const CoreData*>(data_ptr);
```

Comment:
Please don't. Use `memcpy` instead, since it doesn't actually invoke undefined behavior. Ditto below for all usages as a type-punning mechanism below.

Reply:
Fixed.